File: app/src/main/java/com/rupam/notes/Activities/MainActivity.java
package com.rupam.notes.Activities;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.FirebaseDatabase;
import com.rupam.notes.R;

public class MainActivity extends AppCompatActivity {

    private EditText emailIn;
    private EditText passIn;
    private Button signInBtn;
    private Button createAccBtn;
    private FirebaseAuth mAuth;




    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getSupportActionBar().hide();

        emailIn = findViewById(R.id.emailInID);
        passIn = findViewById(R.id.passInID);
        signInBtn = findViewById(R.id.signinBtnID);
        createAccBtn = findViewById(R.id.createAccBtnID);

        mAuth = FirebaseAuth.getInstance();

        signInBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (!TextUtils.isEmpty(emailIn.getText()) &&! TextUtils.isEmpty(passIn.getText())){

                    String getEmail = emailIn.getText().toString();
                    String getPass = passIn.getText().toString();
//                    Checks if the textBoxes are not empty and then passes the values to signIn method
                    signIn(getEmail, getPass);
                }
            }
        });

        //Go to the createAcc activity when createAcc button is clicked
        createAccBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, CreateAccActivity.class));
            }
        });
    }

    //gets the current user and see if it is null (i.e signed in) or not
    @Override
    protected void onStart() {
        super.onStart();
        FirebaseUser currentUser = mAuth.getCurrentUser();
        if(currentUser != null){
            //User is already Signed in
            Toast.makeText(this, "User Signed In", Toast.LENGTH_SHORT).show();
            startActivity(new Intent(MainActivity.this, NotesListActivity.class));
            finish();
        }else{
            //User is not signed in
            Toast.makeText(this, "Please Sign In First", Toast.LENGTH_SHORT).show();
        }
    }

    //The SignIn method to signIn with email and pass
    public void signIn(String email, String pass){
        mAuth.signInWithEmailAndPassword(email, pass)
                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if(task.isSuccessful()){
                            Toast.makeText(MainActivity.this, "Signed In Successfully", Toast.LENGTH_SHORT).show();
                            startActivity(new Intent(MainActivity.this, NotesListActivity.class));
                            finish();
                        }else{
                            Toast.makeText(MainActivity.this, "Sign-In Failed", Toast.LENGTH_SHORT).show();
                        }
                    }
                });
    }
}

File: app/src/main/java/com/rupam/notes/Activities/MainActivity.kt
package com.rupam.notes.Activities

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.content.ContextCompat
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.findNavController
import com.google.firebase.auth.FirebaseAuth
import com.rupam.notes.R

class MainActivity : AppCompatActivity() {

    private lateinit var navController:NavController


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        supportActionBar!!.hide()



        val toolbar:Toolbar = findViewById<Toolbar>(R.id.customToolbar)
        toolbar.title = ""

        setSupportActionBar(toolbar)
        toolbar?.navigationIcon = ContextCompat.getDrawable(this, R.drawable.back_button)

        navController = findNavController(R.id.fragmentContainer)


    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.sign_out_menu, menu)

//        val getAddNewItem:MenuItem = menu!!.findItem(R.id.addNew)
//
//
//        var gt = this.getDrawable(R.drawable.add_new_btn)

        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        when(item.title){

            "Add New" -> navController.navigate(R.id.addNewNoteFragment)
            "Sign Out" -> Log.d("TAG", "onOptionsItemSelected: ")

        }
        return super.onOptionsItemSelected(item)
    }




}